src/packagedcode/go_mod.py

```diff
@@ -0,0 +1,199 @@
+        r'(?P<name>[^\s]*)'
+    ).match
+
+    parse_require = re.compile(
+        r'^require\s'
+        r'.*/(?P<name>[^\s]*)'
+        r'\s'
+        r'(?P<version>[a-z].*)'
+    ).match
+
+    parse_exclude = re.compile(
+        r'^exclude\s'
+        r'.*/(?P<name>[^\s]*)'
+        r'\s'
+        r'(?P<version>[a-z].*)'
+    ).match
+
+    parse_dep_link = re.compile(
+        r'^.*/(?P<name>[^\s]*)'
+        r'\s'
+        r'(?P<version>[a-z].*)'
+    ).match
+
+    def preprocess(self, line):
+        """
+        Return line string after removing commented portion and excess spaces.
+        """
+        if "//" in line:
+            line = line[:line.index('//')]
+        line = line.strip()
+        return line
+
+    def parse_gomod(self, location):
+        """
+        Return a dictionary containing all the important go.mod file data.
+        """
+        with io.open(location, encoding='utf-8', closefd=True) as data:
+            lines = data.readlines()
+
+        gomod_data = {}
+        require = []
+        exclude = []
+
+        for i, line in enumerate(lines):
+            line = self.preprocess(line)
+            parsed_module = self.parse_module(line)
+            if parsed_module:
+                gomod_data['module'] = parsed_module.group('module')
+
+            parsed_module_name = self.parse_module_name(line)
+            if parsed_module_name:
+                gomod_data['name'] = parsed_module_name.group('name')
+                
+            parsed_require = self.parse_require(line)
+            if parsed_require:
+                line_req = [parsed_require.group('name'), parsed_require.group('version')]
+                require.append(line_req)
+
+            parsed_exclude = self.parse_exclude(line)
+            if parsed_exclude:
+                line_req = [parsed_exclude.group('name'), parsed_exclude.group('version')]
+                exclude.append(line_req)
+
+            if 'require' in line and '(' in line:
+                for req in lines[i+1:]:
+                    req = self.preprocess(req)
+                    if ')' in req:
+                        break
+                    parsed_dep_link = self.parse_dep_link(req)
+                    if parsed_dep_link:
+                        line_req = [parsed_dep_link.group('name'), parsed_dep_link.group('version')]
+                        require.append(line_req)
+
+            if 'exclude' in line and '(' in line:
+                for exc in lines[i+1:]:
+                    exc = self.preprocess(exc)
+                    if ')' in exc:
+                        break
+                    parsed_dep_link = self.parse_dep_link(exc)
+                    if parsed_dep_link:
+                        line_exclude = [parsed_dep_link.group('name'), parsed_dep_link.group('version')]
+                        exclude.append(line_exclude)
+
+        gomod_data['require'] = require
+        gomod_data['exclude'] = exclude
+
+        return gomod_data
+
+
+"""
```

Comment:
This should be at the top, not at the bottom.

Reply:
@rpotter12 I think this is still an open item?

Reply:
I will edit this in some time :)